clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java

```diff
@@ -246,11 +228,47 @@ private void recordProducerStats(int producedRecordSize, int producedRecordNum)
     ingestionTask.getHostLevelIngestionStats().recordTotalLeaderRecordsProduced(producedRecordNum);
   }
 
-  protected Put instantiateChunkPut() {
+  protected Put instantiateValueChunkPut() {
+    return new Put();
+  }
+
+  protected Put instantiateRmdChunkPut() {
     return new Put();
   }
 
   protected Put instantiateManifestPut() {
     return new Put();
   }
+
+  private long produceChunksToStoreBufferService(
+      ChunkedValueManifest manifest,
+      ByteBuffer[] chunks,
+      boolean isRmdChunks) throws InterruptedException {
+    long totalChunkSize = 0;
+    for (int i = 0; i < manifest.keysWithChunkIdSuffix.size(); i++) {
+      ByteBuffer chunkKey = manifest.keysWithChunkIdSuffix.get(i);
+      ByteBuffer chunkValue = chunks[i];
+      Put chunkPut;
+      if (isRmdChunks) {
+        chunkPut = instantiateRmdChunkPut();
+        chunkPut.schemaId = AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion();
+        chunkPut.replicationMetadataPayload = chunkValue;
+      } else {
+        chunkPut = instantiateValueChunkPut();
+        chunkPut.putValue = chunkValue;
+        chunkPut.schemaId = AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion();
```

Comment:
There is a duplicate line between `if` scope and `else` scope:
```
        chunkPut.schemaId = AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion();
```
Can we move it out?